canard.c

```diff
@@ -37,15 +37,14 @@
 #define TRANSFER_TIMEOUT_USEC                       2000000
 
 #define TRANSFER_ID_BIT_LEN                         5U
-#define ANON_MSG_DATA_TYPE_ID_BIT_LEN               2U
 
-#define SOURCE_ID_FROM_ID(x)                        ((uint8_t) (((x) >> 0U)  & 0x7FU))
-#define SERVICE_NOT_MSG_FROM_ID(x)                  ((bool)    (((x) >> 7U)  & 0x1U))
-#define REQUEST_NOT_RESPONSE_FROM_ID(x)             ((bool)    (((x) >> 15U) & 0x1U))
+#define SOURCE_ID_FROM_ID(x)                        ((uint8_t) (((x) >> 1U)  & 0x7FU))
+#define SERVICE_NOT_MSG_FROM_ID(x)                  ((bool)    (((x) >> 25U) & 0x1U))
+#define REQUEST_NOT_RESPONSE_FROM_ID(x)             ((bool)    (((x) >> 26U) & 0x1U))
 #define DEST_ID_FROM_ID(x)                          ((uint8_t) (((x) >> 8U)  & 0x7FU))
-#define PRIORITY_FROM_ID(x)                         ((uint8_t) (((x) >> 24U) & 0x1FU))
+#define PRIORITY_FROM_ID(x)                         ((uint8_t) (((x) >> 26U) & 0x1FU))
 #define MSG_TYPE_FROM_ID(x)                         ((uint16_t)(((x) >> 8U)  & 0xFFFFU))
-#define SRV_TYPE_FROM_ID(x)                         ((uint8_t) (((x) >> 16U) & 0xFFU))
+#define SRV_TYPE_FROM_ID(x)                         ((uint8_t) (((x) >> 15U) & 0x1FFU))
 
 #define MAKE_TRANSFER_DESCRIPTOR(data_type_id, transfer_type, src_node_id, dst_node_id)             \
     (((uint32_t)(data_type_id)) | (((uint32_t)(transfer_type)) << 16U) |                            \
```

Comment:
We don't have data type ID anymore. We have *port ID*, which in turn can be either *service ID* or *subject ID*.